configure.ac

```diff
@@ -1929,6 +1935,17 @@ AC_COMPILE_IFELSE(
         pthread_getaffinity_np(pthread_self(), sizeof(cs), &cs);]])],
     [AC_DEFINE([HAS_BSD_GETAFFINITY_NP])])])
 
+## ZSTD compression library
+
+AS_IF([test x"$with_zstd" = "xyes"],
+  AS_IF([pkg-config --atleast-version 1.4 libzstd 2>/dev/null],
```

Comment:
Just noting that I think this adds a new dependency on `pkg-config` for the OCaml compiler build, but that I think it's probably fine. Although, at a glance on most common distros zstd is pretty standalone, and so just using AC_SEARCH_LIBS might be sufficient and avoid this dependency.

Reply:
Right.  I'm not fixated on using pkg-config, and the quick-and-dirty test above just fails silently if pkg-config is not available.  Moreover, the test will certainly have to be reworked to handle the Windows ports.

> Although, at a glance on most common distros zstd is pretty standalone

Linux distros, yes.  For macOS, zstd is in bizarre Homebrew or MacPorts locations, and I don't know of any better way to find those locations than pkg-config.

Reply:
Also, because the zstd streaming API is in constant flux, we need to check for version >= 1.4, version 1.3 (as found in Ubuntu 18.04 LTS for instance) doesn't work.  Again, pkg-config makes it trivial to do this check, but AC_SEARCH_LIBS will need a bit more work.

Reply:
I pushed an improved zstd detection code: it tries pkg-config first, and if it fails it then uses `AC_CHECK_LIBS` and `AC_CHECK_DECL` to see if zlib is in the standard location and defines the functions we need.  On our CI most of the machines have pkg-config, and those who don't have no zstd library or an outdated version, so the new code is not fully exercised.